src/php/Lang/Set.php

```diff
@@ -0,0 +1,86 @@
+<?php
+
+namespace Phel\Lang;
+
+use ArrayAccess;
+use Countable;
+use InvalidArgumentException;
+use Iterator;
+use Phel\Printer;
+
+class Set extends PhelArray implements ArrayAccess, Countable, Iterator, ICons, ISlice, ICdr, IRest, IPop, IRemove, IPush, IConcat
+{
+
+    /**
+     * Constructor
+     *
+     * @param mixed[] $data A list of all values
+     */
+    public function __construct(array $data)
+    {
+        parent::__construct([]);
+        $this->setData($data);
+    }
+
+    /**
+     * Create a new array.
+     *
+     * @param mixed[] $values A list of all values
+     *
+     * @return Set
+     */
+    public static function create(...$values): Set
+    {
+        return new Set($values);
+    }
+
+    public function push($x): IPush
+    {
+        if (!in_array($x, $this->data)) {
```

Comment:
This is not a good idea, because it is slow. `in_array` has the complexity `O(n)`. But checking if a value is in a set must be `O(1)`. 

You have to do this the following way.
1. Compute the hash of the value (see `Table` class)
2. Store the hash as key of the PHP Array and the value as the value of the PHP Array.

The push method looks like this:

```
public function push($x): IPush
{
    $hash = $this->computeHash($x);
    $this->data[$hash] = $x; // Don't need to check if $x is already there, just override.

   return $this;
}
```

Afterwards, you probably can simplify the methods for intersection, union and diff.